cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -1373,14 +1368,27 @@
 
 ;;;; Helpers for determining the register class of a value type ;;;;;;;;;;;;;;;;
 
+(type RegisterClass
+      (enum
+        (Gpr (single_register bool))
+        (Xmm)))
+
+(decl type_register_class (RegisterClass) Type)
+(extern extractor type_register_class type_register_class)
+
 (decl is_xmm_type (Type) Type)
-(extern extractor is_xmm_type is_xmm_type)
+(extractor (is_xmm_type ty) (and (type_register_class (RegisterClass.Xmm)) ty))
```

Comment:
Rather than the `(and subpat var)` idiom here, could we write either `(and var subpat)`, or preferably `var @ subpat`? It's a little hard for me to visually parse what's happening here otherwise. (Likewise for the three cases below.)

Reply:
We can't write `var @ subpat` because ISLE doesn't currently support that for the RHS of an extractor definition. I standardized on the `(and pat var)` form to ensure that these newly introduced `and` patterns wouldn't run afoul of the `and` order bug that we discussed in the meeting today. Happy to reorder them once we fix that bug, as I like the `(and var pat)` form more as well.

Reply:
Ah, yeah, we should fix that but for now this is fine then.